pkg/agent/controller/egress/egress_controller.go

```diff
@@ -534,8 +699,14 @@ func (c *EgressController) realizeEgressIP(egressName, egressIP string) (uint32,
 			}
 			ipState.ruleInstalled = true
 		}
+		if err := c.installPolicyRoute(ipState, subnetInfo); err != nil {
+			return 0, fmt.Errorf("error installing policy route for IP %s: %v", ipState.egressIP, err)
+		}
 	} else {
 		// Ensure datapath is uninstalled properly.
+		if err := c.uninstallPolicyRoute(ipState); err != nil {
```

Comment:
I assume `installPolicyRoute/uninstallPolicyRoute` are idempotent, so you need not another boolean like `flowsInstalled`, `ruleInstalled`?

Reply:
`ipState.subnetInfo` is kind of "subnetInfoInstalled", it's just not a boolean but a struct because it's not just true or false. It's checked inside `installPolicyRoute/uninstallPolicyRoute` to save repeated code on the caller side.